src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosExceptionExtensions.cs

```diff
@@ -54,5 +55,16 @@ public static class CosmosExceptionExtensions
                 ? subStatusCode
                 : (int?)null;
         }
+
+        /// <summary>
+        /// Determines if the error is due to a client customer-managed key error
+        /// </summary>
+        /// <param name="exception">The exception object</param>
+        /// <returns>True iff the error is due to client CMK setting.</returns>
+        public static bool IsCmkClientError(this CosmosException exception)
+        {
+            return exception.StatusCode == HttpStatusCode.Forbidden
+                && (Enum.IsDefined(typeof(KnownCosmosDbCmkSubStatusValueClientIssue), exception.SubStatusCode) || exception.SubStatusCode == 3);
```

Comment:
For a substatus code, I expected values from KnownCosmosDbCmkSubStatusValueClientIssue.  However, when testing this scenario, I noticed that Cosmos DB is returning substatus=3.  I think this is an issue and filed incident 217300148.  However, to make this scenario work work today, I have allowed for this unexpected status code.

Reply:
Per the incident, there are no current plans to update this behavior.  So, I will keep the clause to handle substatus=3

Reply:
Should we add it as a value of the enum if it is "expected"? Either way I think a comment in the code explaining why this value is here would be valuable.

Reply:
I would like to add it to the enum of expected values, but it's not one that is actually expected, per documentation.  It's one that has been observed, and then acknowledged by Cosmos DB.  So, I have added a comment here.  Let me know if you prefer some other wordsmithing.